### PR TITLE
[KED-1732] Add json.JSONDataSet

### PR DESCRIPTION
**Pull request by tamsanh**
## Description
I and a few others missed having JSON as a built in dataset, so here is one that hopefully fulfills that need.

## Development notes
I created a JSONDataSet based off of the YAMLDataSet. The only difference between the two is that JSONDataSet removes support for pandas dataframes. There's some oddity with reloading a dataframe from a JSON file, so opted to remove it instead.

## Checklist

- [x] Read the [contributing](../CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](../RELEASE.md) file
- [x] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: tests/extras/datasets/json/test_json_dataset.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        "save_version", ["2019-01-02T00.00.00.000Z"], indirect=True
+    )
+    def test_save_version_warning(
+        self, versioned_json_data_set, load_version, save_version, dummy_data
+    ):
+        """Check the warning when saving to the path that differs from
+        the subsequent load path."""
+        pattern = (
+            r"Save version `{0}` did not match load version `{1}` "
+            r"for JSONDataSet\(.+\)".format(save_version, load_version)
```
Comment (lorenabalan) [inline tests/extras/datasets/json/test_json_dataset.py:198]: You can `f-string`ify this now.

**Comment by mzjp2** [comment]
I think you'll also need to add your dataset to the list here: https://github.com/quantumblacklabs/kedro/blob/develop/docs/source/05_api_docs/kedro.extras.datasets.rst (in correct alphabetical order please 🙇‍♂️) and here https://github.com/quantumblacklabs/kedro/blob/develop/docs/source/04_user_guide/08_advanced_io.md#supported-datasets and then you should be good to go :)

**Comment by mzjp2** [comment]
Oh, and can you update `setup.py` with the required dependencies?

**Comment by lorenabalan** [comment]
> Oh, and can you update `setup.py` with the required dependencies?

I think `json` is part of the standard library so it should be fine. We don't provide a key for `text.TextDataSet` either.

**Comment by mzjp2** [comment]
> > Oh, and can you update `setup.py` with the required dependencies?
> 
> I think `json` is part of the standard library so it should be fine. We don't provide a key for `text.TextDataSet` either.

My bad, ignore me 🙏 - was looking at some other datasets and this popped into my head, should have looked more in-depth 🤦

**Pull request merged**